runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -379,6 +379,13 @@ else if (cballsLeft > 5)
 		{
 			return Color.orange;
 		}
+		else if (cballsLeft <= config.ammoAmount())
```

Comment:
Whenever `cballsLeft` is at or below the configured threshold, this code would try to send a notification once per render frame wouldn't it? There is no flag to indicate that it has been sent once already, so it would continue doing this indefinitely until the count goes back above the threshold.

Also, if both `config.ammoAmount()` and `config.notifyAmmoLeft()` are maintained, their if conditions should be combined.

Finally, why is this an `else if` statement within the `getStateColor()` method? This method should be used exclusively for returning the color of the text and not have side effects like this. This code should probably be run in the `GameTick` subscriber instead.